content/swiftui/concepts/animation/animation.md

```diff
@@ -0,0 +1,104 @@
+---
+Title: 'Animation'
+Description: 'Produce seamless visual updates in response to state changes.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'SwiftUI'
+  - 'SwiftUI ViewModifiers'
+  - 'iOS'
+CatalogContent:
+  - 'learn-iOS-course'
+  - 'path/iOS-path'
+---
+
+**Animation** is the ability to add smooth view changes to a view by changing the state of the view. There are two types of animation in SwiftUI: implicit and explicit animation. 
+
+It's a very powerful tool in SwiftUI that enhances the user experience of an app by making it more interactive. Custom views can be animated by conforming to the `Animatable` protocol, which tells SwiftUI about the value to animate.
+
+## Implicit Animation with `.animation()`
+
+To be able to add animation to a view, three things are required:
+
+1. A variable that changes to activate the animation. 
+2. Flow control to manage the animation.
+3. The actual animation; use of the `.animation()` modifier on the desired view to be animated.
+
+These components are essential to build animation implicitly or explicitly on a view. In the next example, implicit animation will be used:
+
+```swift
+struct ImplicitAnimation: View {
+    @State var isanimation = false // @State variable that changes to activate the animation.
+    
+    var body: some View {
+        Button {
+            isanimation.toggle()
+        } label: {
+            Text( isanimation ? "Hello, World!" : "press me") // Ternary used to change the view based on the value of the isanimation.
+                .foregroundColor(.white)
+                .padding(.all)
+                .background( isanimation ? Color.black : Color.red) // Ternary operator used to change the view based onn the value of the isanimation.
+                .animation(.default, value: isanimation) // This is the implicit animation. This gets activated when isanimation changes in value.
+        }
+    }
+}
+```
+
+This is the result of the code above:
+
+![This is a gif that shows the effect of Implicit animation on the view.](https://raw.githubusercontent.com/Codecademy/docs/main/media/Implicit-Animation-SwiftUI.gif)
+
+In this simple animation where the background and the text of the button changes depending on the value of the `isanimation` variable. When the button is pressed, `isanimation` changes, causing the implicit animation to change the text and background of the button. As the name implies, SwiftUI creates and manages implicit animations whenever specific situations or user interface state changes take place, which means developers don't have full control over the animations. With explicit animation, developers have full control over them.
```

Comment:
[Proofreading] The first half of the sentence needs clarity. Consider using the suggestion I placed below: 
```suggestion
In this simple animation, the background and the text of the button change depending on the value of the `isanimation` variable. When the button is pressed, `isanimation` changes, causing the implicit animation to change the text and background of the button. As the name implies, SwiftUI creates and manages implicit animations whenever specific situations or user interface state changes take place, which means developers don't have full control over the animations. With explicit animation, developers have full control over them.
```